JLio.Commands/Advanced/Compare.cs

```diff
@@ -0,0 +1,482 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using JLio.Commands.Advanced.Models;
+using JLio.Commands.Advanced.Settings;
+using JLio.Commands.Builders;
+using JLio.Commands.Logic;
+using JLio.Core;
+using JLio.Core.Contracts;
+using JLio.Core.Models;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace JLio.Commands.Advanced
+{
+    public class Compare : CommandBase
+    {
+        private IExecutionOptions options;
+
+        [JsonProperty("firstPath")]
+        public string FirstPath { get; set; }
+
+        [JsonProperty("resultPath")]
+        public string ResultPath { get; set; }
+
+        [JsonProperty("secondPath")]
+        public string SecondPath { get; set; }
+
+        [JsonProperty("settings")]
+        public CompareSettings Settings { get; set; }
+
+        public override ValidationResult ValidateCommandInstance()
+        {
+            var result = new ValidationResult {IsValid = true};
+            if (string.IsNullOrWhiteSpace(FirstPath))
+            {
+                result.ValidationMessages.Add($"FirstPath property for {CommandName} command is missing");
+                result.IsValid = false;
+            }
+
+            if (string.IsNullOrWhiteSpace(SecondPath))
+            {
+                result.ValidationMessages.Add($"SecondPath property for {CommandName} command is missing");
+                result.IsValid = false;
+            }
+
+            if (string.IsNullOrWhiteSpace(ResultPath))
+            {
+                result.ValidationMessages.Add($"ResultPath property for {CommandName} command is missing");
+                result.IsValid = false;
+            }
+
+            if (Settings == null)
+            {
+                result.ValidationMessages.Add($"Settings property for {CommandName} command is missing");
+                result.IsValid = false;
+            }
+
+            return result;
+        }
+
+        public override JLioExecutionResult Execute(JToken dataContext, IExecutionOptions executionOptions)
+        {
+            options = executionOptions;
+            var compareResults = new CompareResults();
+            var validationResult = ValidateCommandInstance();
+            if (!validationResult.IsValid)
+            {
+                validationResult.ValidationMessages.ForEach(i =>
+                    options.Logger?.Log(LogLevel.Warning, Constants.CommandExecution, i));
+                return new JLioExecutionResult(false, dataContext);
+            }
+
+            foreach (var source in options.ItemsFetcher.SelectTokens(FirstPath, dataContext))
+            foreach (var target in options.ItemsFetcher.SelectTokens(SecondPath, dataContext))
```

Comment:
please do proper indentation, or use curly brackets. puting the 2 foreach statements on the same indentation level is misleading here

Reply:
i will fix this later. it is the automatic cleanup that mess things up. i have to find out how to set the proper settings. after that it will be fixed automatically